dataland-internal-storage/src/main/kotlin/db/migration/V9__MigrateRefactoredDataPointClassesEuTaxNonFinancials.kt

```diff
@@ -0,0 +1,89 @@
+package db.migration
+
+import db.migration.utils.DataTableEntity
+import db.migration.utils.MigrationHelper
+import db.migration.utils.getOrJavaNull
+import db.migration.utils.migrateCompanyAssociatedDataOfDatatype
+import org.flywaydb.core.api.migration.BaseJavaMigration
+import org.flywaydb.core.api.migration.Context
+import org.json.JSONObject
+
+/**
+ * This migration script migrates the eu taxonomy non financials data to the various new class structures for
+ * data points
+ */
+class V9__MigrateRefactoredDataPointClassesEuTaxNonFinancials : BaseJavaMigration() {
+    override fun migrate(context: Context?) {
+        migrateCompanyAssociatedDataOfDatatype(
+            context,
+            "eutaxonomy-non-financials",
+            this::migrateRefactoredDataPointClasses,
+        )
+    }
+
+    private val oldToNewFieldNamesForReports = mapOf(
```

Comment:
The "totalAmount" field change from Datapoint with unit to a currency data point. That seems not to be covered by the migration.

Reply:
unit to currency has been added, rest is obsolete as discussed